pandas/tests/io/formats/style/test_highlight.py

```diff
@@ -181,9 +181,20 @@ def test_highlight_between_inclusive(styler, inclusive, expected):
 )
 def test_highlight_quantile(styler, kwargs):
     expected = {
+        (0, 1): [("background-color", "yellow")],
         (2, 0): [("background-color", "yellow")],
         (2, 1): [("background-color", "yellow")],
     }
+    if styler.data.dtypes["B"] != "Int64":
+        expected.pop((0, 1))
+    else:
+        if kwargs.get("axis", -1) is None:
+            expected.pop((0, 1))
+        elif kwargs.get("q_left", -1) == 0:
+            expected.pop((0, 1))
+        elif "subset" in kwargs:
+            expected.pop((0, 1))
```

Comment:
```suggestion
    if kwargs.get("axis", -1) == 0 and styler.data.dtypes["B"] == "Int64":
        # pd.NA is interpreted as zero value in the quantile calculation meaning
        # of values {1, <NA>, 10}, {1, 10} are returned within the [0.5, 1] quantile.
        expected[(0, 1)] = [("background-color", "yellow")]
```

Reply:
[[ the alternative fix is to change the pytest fixture column values from {1, pd.NA, 2} to {1, pd.Na, 3}, since then the value 1 is not caught within the [0.5, 1] quantile where pd.NA = 0, and conveniently all other tests in the module still pass ]]

Reply:
Do you think the trouble is in the quantile behavior or in the Styler?

Reply:
OK, further investigation shows that this is a rounding issue when preserving the dtype in the quantile function. With

```python
>>> df
      A     B
0     0     1
1  <NA>  <NA>
2    10     2
```

Then calling `df.quantile(axis=0, numeric_only=False, q=[0.5, 1], interpolation="linear")` gives under `dtype="Float64"`:

```python
        A    B
0.5   5.0  1.5
1.0  10.0  2.0
dtypes=Float64
```

Whereas under this PR using `dtype="Int64"` the values are:

```python
      A  B
0.5   5  1
1.0  10  2
dtypes=Int64
```

The value of 1 is included in the second, probably due to calling `int(1.5)`, and so the styler highlighting is changed for cell index (0,1).

Prior to this PR the dtype of column B was upcast from "Int64" to "float64" and the leftmost quantile was calculated as 1.5. Since the interpolation style is linear and that is specifically documented to be fraction, I think it would be worth expanding the docs how dtypes are preserved and can affect calculations, or considering upcasting to "Float64".

It is possible to fix all tests in Styler by changing the fixture's column B to have `4` instead of `2` so that the rounding of the 50% quantile at `2.5` to `2` still excludes cell (0,1).

Reply:
thanks @attack68 I think I've got a handle on whats going wrong, and it tentatively looks like its in the quantile code.  ill try the fix i have in mind and let you know if that doesnt work out